src/utils/miconoss/Newton.C

```diff
@@ -0,0 +1,176 @@
+#include "Newton.h"
+
+Newton::Newton(NLSystem *_nlsys, NLSystemVars *sys_vars, const double tolerance,
+               const uint _max_iter, const miconossmath::normtype normtype)
+    : _nlsys(_nlsys), _sys_vars(sys_vars), _tolerance(tolerance),
+      _max_iter(_max_iter), _n_eq(_nlsys->getNx()),
+      _n_lm(_nlsys->getNmultipliers()), _n_total(_n_eq + _n_lm), _dx(_n_total),
+      _normtype(normtype) {}
+
+int Newton::computeNewtonStep(const vecD &R, const matrixD &J) {
+  int ierr = miconossmath::solveAxb(J, R, _n_total, _dx);
+
+  for (uint i = 0; i < _n_total; i++)
+    _dx[i] = -_dx[i];
+  return ierr;
+}
+
+int Newton::customSubstepInterruption(NLSystemParameters *const sysparams,
+                                      bool &custom_interruption_flag) {
+  custom_interruption_flag = false;
+  return 0;
+}
+
+int Newton::solve(vecD &lm, matrixD &J, bool &converged,
+                  const bool auto_scale_equation) {
+  converged = false;
+  double err = 1e6;
+  int ierr = 0;
+  uint it = 0;
+  if (auto_scale_equation)
+    _nlsys->updateEquationScaling();
+
+  _nlsys->updateEquationConstants();
+  vecD R = _nlsys->assembleR(lm);
+  ierr = miconossmath::norm(R, _normtype, err);
+  if (ierr == 0) {
```

Comment:
Get rid of commented out statements or hide behind a debug flag.

Reply:
done